app/controllers/MonthlyPaymentAmountController.scala

```diff
@@ -162,4 +161,23 @@ object MonthlyPaymentAmountController {
       key -> Forms.of(amountOfMoneyFormatter(minimumAmount.inPounds > _, maximumAmount.inPounds < _))
     )(identity)(Some(_))
   )
+
+  private def upfrontPaymentAnswersFromJourney(journey: Journey.AfterRetrievedAffordabilityResult): UpfrontPaymentAnswers = {
```

Comment:
Sorry, this is probably the worst bit about the pr, we can't use the old way of `j.into[Journey.AfterUpfrontPaymentAnswers]` as chimney complains about ambiguity, same for the similar approach in TtpService below

Reply:
we may be able to have a shorter pattern match at least here maybe - does something like
```scala
journey match {
  case j: Journey.AfterUpfrontPaymentAnswers => j.upfrontPaymentsAnswers
  case j: Journey.BeforeUpfrontPaymentAnswers => // error
```
work? Though the second case is annoying cos we don't actually have a way of reaching it atm

Reply:
updated